camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/DslYamlTest.java

```diff
@@ -0,0 +1,655 @@
+    }
+
+    @Test(groups="WIP") // config accepts strings only, no suppliers
+    public void testDslConfigWithDeferredArg() throws Exception {
+        final Entity app = createAndStartApplication(
+                "services:",
+                "- type: " + BasicApplication.class.getName(),
+                "  brooklyn.config:",
+                "    source: myvalue",
+                "    configName: source",
+                "    dest: $brooklyn:config(config(\"configName\"))");
+        assertEquals(getConfigEventually(app, DEST), "myvalue");
+    }
+
+    @Test(groups="WIP") // config accepts strings only, no suppliers
+    public void testDslConfigOnEntityWithDeferredArg() throws Exception {
+        final Entity app = createAndStartApplication(
+                "services:",
+                "- type: " + BasicApplication.class.getName(),
+                "  brooklyn.config:",
+                "    entityName: sourceEntity",
+                "    configName: source",
+                "    dest: $brooklyn:entity(config(\"entityName\")).config(config(\"configName\"))",
+                "  brooklyn.children:",
+                "  - type: " + BasicEntity.class.getName(),
+                "    id: sourceEntity",
+                "    brooklyn.config:",
+                "      source: myvalue");
+        assertEquals(getConfigEventually(app, DEST), "myvalue");
+    }
+
+    @Test
+    public void testDslAttributeWhenReady() throws Exception {
+        final Entity app = createAndStartApplication(
+                "services:",
+                "- type: " + BasicApplication.class.getName(),
+                "  brooklyn.initializers:",
+                "  - type: org.apache.brooklyn.core.sensor.StaticSensor",
+                "    brooklyn.config:",
+                "      name: source",
+                "      static.value: myvalue",
+                "  brooklyn.config:",
+                "    dest: $brooklyn:attributeWhenReady(\"source\")");
+        assertEquals(getConfigEventually(app, DEST), "myvalue");
+    }
+
+    @Test
+    public void testDslAttributeWhenReadyOnEntity() throws Exception {
+        final Entity app = createAndStartApplication(
+                "services:",
+                "- type: " + BasicApplication.class.getName(),
+                "  brooklyn.config:",
+                "    dest: $brooklyn:entity(\"sourceEntity\").attributeWhenReady(\"source\")",
+                "  brooklyn.children:",
+                "  - type: " + BasicEntity.class.getName(),
+                "    id: sourceEntity",
+                "    brooklyn.initializers:",
+                "    - type: org.apache.brooklyn.core.sensor.StaticSensor",
+                "      brooklyn.config:",
+                "        name: source",
+                "        static.value: myvalue");
+        assertEquals(getConfigEventually(app, DEST), "myvalue");
+    }
+
+    @Test(groups="WIP") // config accepts strings only, no suppliers
```

Comment:
Cut-n-paste - think this comment should be `attributeWhenReady accepts strings only, no suppliers`, also on next method